### PR TITLE
Clone or pull projects on batch command

### PR DESCRIPTION
**Pull request by B3rs**
Currently the batch sub command is not pulling the repositories when, looking at his semantics I think it should
This MR adds a clone or pull function in the main loop to pull the repositories before executing the checks, ensuring their presence.

### CODE REVIEW

**Comment by victorboissiere** [comment]
#sre

File: cmd/batch.go [hunk 1]
```diff
@@ -51,3 +62,14 @@ func batchCmd(configPath, logLevel, logFormat *string) *cobra.Command {
 
 	return cmd
 }
+
+func cloneOrPull(git *git.Git, project project.Project) error {
```
Comment (victorboissiere) [inline cmd/batch.go:66]: Do we want to clone? Some projects have >50GB. It is risky IMHO (or needs to update the chart deployment for a specific config).
Maybe we can just fetch the latest commit (like CI behavior by default). Wdyt?

Comment (B3rs) [reply cmd/batch.go:66]: It is a good idea, I just reused what we do for the daemon command for now

Comment (B3rs) [reply cmd/batch.go:66]: I see that shallow clone is disabled [here](https://github.com/qonto/standards-insights/blob/c2c8d881ead843deb6e5a97b98140673993c1bc9/internal/git/git.go#L60), do you remember why @mcorbin ?

Comment (mcorbin) [reply cmd/batch.go:66]: I wasn't sure about the impact: https://github.com/CocoaPods/CocoaPods/issues/4989#issuecomment-193772935

**Review by mcorbin — APPROVED** [review]

**Pull request merged**